src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs

```diff
@@ -82,24 +83,24 @@ public static Summary RunSource(string source, IConfig config = null)
         [MethodImpl(MethodImplOptions.NoInlining)]
         private static Summary RunWithDirtyAssemblyResolveHelper(Type type, IConfig config, string[] args)
             => (args == null
```

Comment:
All the checks are performed only when `args` are null`. We should rather have something like:

```cs
$SomeType $SomeMethod($arguments, string[] args)
{
     Validate($arguments); // throws when something is wrong

     if (args is null)
        BenchmarkSwitcher...
     else
        BenchmarkRunner...
}

Reply:
Done.

I wanted to mimic the behavior of BenchmarkSwitcher (BenchmarkRunner.Run used it when `args != null`).

But sometimes it would write a nice message and then throw exception (see test cases 6,7,8 in my comment)